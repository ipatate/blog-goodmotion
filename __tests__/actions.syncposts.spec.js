const actions = require('../src/actions/actions');
const nock = require('nock');

const page = 1;

describe('test syncPosts function', () => {

  it('syncPosts no error', async (done) => {
        nock('https://api.tvmaze.com/')
        .get(`/search/shows?q=girls&page=${page}`)
        .reply(200, {});

        try {
          const data = await actions.syncPosts();
          expect(data).toMatchSnapshot();
          done()
        }catch (e) {
          expect(e).toBeUndefined()
        }
  })

  it('syncPosts error 500', async (done) => {
        nock('https://api.tvmaze.com/')
        .get(`/search/shows?q=girls&page=${page}`)
        .replyWithError({'message': 'Error server', 'code': '500'})

        try {
          await actions.syncPosts();
        }catch (e) {
          expect(e).toMatchSnapshot();
          done()
        }
  })


  it('syncPosts error 404', async (done) => {
        nock('https://api.tvmaze.com/')
        .get(`/search/shows?q=girls&page=${page}`)
        .reply(404)

        try {
          await actions.syncPosts();
        }catch (e) {
          expect(e).toMatchSnapshot();
          done()
        }
  })

});
