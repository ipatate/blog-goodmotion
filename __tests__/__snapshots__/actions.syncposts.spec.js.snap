// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test syncPosts function syncPosts error 404 1`] = `[FetchError: invalid json response body at https://api.tvmaze.com/search/shows?q=girls&page=1 reason: Unexpected end of JSON input]`;

exports[`test syncPosts function syncPosts error 500 1`] = `[FetchError: request to https://api.tvmaze.com/search/shows?q=girls&page=1 failed, reason: Error server]`;

exports[`test syncPosts function syncPosts no error 1`] = `undefined`;
